lib/spells/effects/Catapult.cpp

```diff
@@ -57,38 +57,26 @@ bool Catapult::applicable(Problem & problem, const Mechanics * m) const
 	return !attackableBattleHexes.empty() || m->adaptProblem(ESpellCastProblem::NO_APPROPRIATE_TARGET, problem);
 }
 
-void Catapult::apply(ServerCallback * server, const Mechanics * m, const EffectTarget & /* eTarget */) const
+void Catapult::apply(ServerCallback * server, const Mechanics * m, const EffectTarget & eTarget) const
 {
-	//start with all destructible parts
-	static const std::set<EWallPart> potentialTargets =
-	{
-		EWallPart::KEEP,
-		EWallPart::BOTTOM_TOWER,
-		EWallPart::BOTTOM_WALL,
-		EWallPart::BELOW_GATE,
-		EWallPart::OVER_GATE,
-		EWallPart::UPPER_WALL,
-		EWallPart::UPPER_TOWER,
-		EWallPart::GATE
-	};
-
-	assert(potentialTargets.size() == size_t(EWallPart::PARTS_COUNT));
+	if(isSeparate)
+		applySeparate(server, m, eTarget); // Like catapult shots
+	else
+		applyMerged(server, m); // Like earthquake
```

Comment:
Please rename for consistency:
1) Catapult shot: applyTargeted
2) Earthquake spell applyMassive (this is the term used by all non-targetable spells)
3) Rename config key from "separate" to "massive" (or "isMassive", not sure which form is more common in spell effects)

Reply:
isTargeted perhaps? Because now we checks for targeted instead.

Reply:
In code there are basically 0 hits for "targeted" but quite a lot for "massive".

So I'd rather use "massive" and adapt this check instead.

Reply:
@IvanSavenko massive is not used in json too. But will look for it.

Reply:
For consistency we can try using range here.
Earthquake will have "X" range (which in code translates to massive)
Ability will have "0" range (single target).

However not sure whether this is a good idea - other ranges likely won't work here.

Reply:
@IvanSavenko the problem is than random catapult shots and NO_LOCATION spells are both set single target with invalid hex, and I want to be able to differentiate this effects.

Reply:
@IvanSavenko done with builtin isMassive check. Should work now.